File: ruby/bob/bob.rb
=begin
Write your code for the 'Bob' exercise in this file. Make the tests in
`bob_test.rb` pass.

To get started with TDD, see the `README.md` file in your
`ruby/bob` directory.
=end

class Bob
  def self.hey(say)
    say = say.strip
    return "Calm down, I know what I'm doing!" if question(say) && shouting?(say)
    return 'Sure.' if question(say)
    return 'Whoa, chill out!' if shouting?(say)
    return 'Fine. Be that way!' if say.gsub(/\s/, '').empty?

    'Whatever.'
  end

  def self.question(say)
    say.end_with?('?')
  end

  def self.shouting?(say)
    !say.scan(/[a-zA-Z]+/).empty? && say == say.upcase
  end
end

File: javascript/bob/bob.js
//
// This is only a SKELETON file for the 'Bob' exercise. It's been provided as a
// convenience to get you started writing code faster.
//

export const hey = (message) => {
  message = message.trim()
  if (!message) {
    return 'Fine. Be that way!'
  }
  if (isQuestion(message) && isYell(message)) {
    return "Calm down, I know what I'm doing!"
  }
  if (isQuestion(message)) {
    return "Sure."
  }
  if (isYell(message)) {
    return "Whoa, chill out!"
  }
  return "Whatever.";
};

function isQuestion(message) {
  return message.endsWith("?")
}

function isYell(message) {
  if (!/[a-z]/i.test(message)) {
    return false
  }
  return message === message.toUpperCase()
}
